include/base.php

```diff
@@ -136,17 +144,66 @@ public function load_strings_translations( $locale = '' ) {
 	 * Applied only if Polylang is active on the new blog.
 	 *
 	 * @since 1.5.1
+	 * @since 3.5 Accept now `$context` parameter.
 	 *
-	 * @param int $new_blog_id  New blog ID.
-	 * @param int $prev_blog_id Previous blog ID.
+	 * @param int    $new_blog_id  New blog ID.
+	 * @param int    $prev_blog_id Previous blog ID.
+	 * @param string $context      Additional context.
 	 * @return void
 	 */
-	public function switch_blog( $new_blog_id, $prev_blog_id ) {
+	public function switch_blog( $new_blog_id, $prev_blog_id, $context ) {
 		if ( $this->is_active_on_new_blog( $new_blog_id, $prev_blog_id ) ) {
 			$this->options = get_option( 'polylang' ); // Needed for menus.
-			remove_action( 'pre_option_rewrite_rules', array( $this->links_model, 'prepare_rewrite_rules' ) );
 			$this->links_model = $this->model->get_links_model();
 		}
+
+		if ( 'restore' === $context ) {
+			$this->restore_prepare_rewrite_rules_actions();
+		} else {
+			$this->remove_prepare_rewrite_rules_actions();
+		}
```

Comment:
This should not be needed but replaced by `remove_all_actions( 'pll_prepare_rewrite_rules' )` which removes all rewrite rules filters related to the old site. The new links mode instance on the new site will fire an action which will add the filters related to the new site. Restoring the current site is not a spacial case as the same process applies.

Reply:
Indeed, this is stil a WIP. Since the test doesn't behave as I expected at the moment ...